mozjs/src/jsimpls.rs

```diff
@@ -380,20 +381,117 @@ impl<T> JS::Rooted<T> {
 }
 
 impl JS::ObjectOpResult {
+    pub fn ok(&self) -> bool {
+        assert_ne!(self.code_, JS::ObjectOpResult_SpecialCodes::Uninitialized as usize);
+        self.code_ == JS::ObjectOpResult_SpecialCodes::OkCode as usize
+    }
+
     /// Set this ObjectOpResult to true and return true.
     pub fn succeed(&mut self) -> bool {
         self.code_ = JS::ObjectOpResult_SpecialCodes::OkCode as usize;
         true
     }
 
+    pub fn fail(&mut self, code: JSErrNum) -> bool {
+        assert_ne!(self.code_, JS::ObjectOpResult_SpecialCodes::OkCode as usize);
+        self.code_ = code as usize;
+        true
+    }
```

Comment:
Let's just expose this API, which is a bit more flexible than exposing a method for each type of failure?

Reply:
I don't have any problems with this, but my initial intention with this PR was porting over the C++ methods on it, not necessarily improving/changing it.

Reply:
Okay. I see now that this is a mirror of the C++ interface. That makes sense to me!